azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_incomingwebhook.go

```diff
@@ -0,0 +1,61 @@
+package serviceendpoint
+
+import (
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointIncomingWebhook schema and implementation for incoming webhook service endpoint resource
+func ResourceServiceEndpointIncomingWebhook() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointIncomingWebhook, expandServiceEndpointIncomingWebhook)
+	r.Schema["webhook_name"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		DefaultFunc: schema.EnvDefaultFunc("AZDO_INCOMING_WEBHOOK_SERVICE_CONNECTION_WEBHOOK_NAME", nil),
+		Description: "The name of the WebHook.",
+	}
+	r.Schema["secret"] = &schema.Schema{
+		Type:             schema.TypeString,
+		Optional:         true,
+		DefaultFunc:      schema.EnvDefaultFunc("AZDO_INCOMING_WEBHOOK_SERVICE_CONNECTION_SECRET", nil),
+		Description:      "Optional secret for the webhook. WebHook service will use this secret to calculate the payload checksum.",
+		Sensitive:        true,
+		DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+	}
+	secretHashKey, secretHashSchema := tfhelper.GenerateSecreteMemoSchema("secret")
+	r.Schema[secretHashKey] = secretHashSchema
```

Comment:
`secretHashKey` can be removed, does not do much help there

Reply:
Can you please explain why? I'm new to golang and provider implementation and did note that other service endpoint implementations do this for sensitive values.

Reply:
this function/hashkey  is used to prevent sensitive  data print to console etc. , it is a wrapper for the sensitive data.

Reply:
Okay, thought the secret is also sensitive data like the password here: https://github.com/microsoft/terraform-provider-azuredevops/blob/main/azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_generic.go#L35

Reply:
Latest Terraform will prevent print sensitive data to console etc. by default,  hash the sensitive data is not much help now. Also the `bcrypt` has a 72 length limit.

Reply:
ok, removed the secretHashKey